### PR TITLE
Tweak xhr error handling

### PR DESCRIPTION
**Pull request by SimonWoolf**
Not sure whether bundling the errorEvent and the current statusText in there will actually be useful, but, shrug

### CODE REVIEW

**Comment by mattheworiordan** [comment]
Looks good, but I think needs @paddybyers's input for this one

File: browser/lib/transport/xhrrequest.js [hunk 1]
```diff
@@ -122,19 +122,19 @@ var XHRRequest = (function() {
 		for(var h in headers)
 			xhr.setRequestHeader(h, headers[h]);
 
-		var onerror = xhr.onerror = function(err) {
-			err.code = 80000;
-			self.complete(err);
+		var errorHandler = function(errorEvent, message, code, statusCode) {
+			var errorMessage = message + ", errorEvent was " + Utils.inspect(errorEvent) + ", current statusText is " + self.xhr.statusText;
```
Comment (paddybyers) [inline browser/lib/transport/xhrrequest.js]: Single quotes please everywhere.

File: browser/lib/transport/xhrrequest.js [hunk 2]
```diff
@@ -122,19 +122,19 @@ var XHRRequest = (function() {
...
+			Logger.logAction(Logger.LOG_ERROR, 'Request.on' + errorEvent.type + '()', errorMessage);
+			self.complete(new ErrorInfo(errorMessage, code, statusCode));
 		};
-		xhr.onabort = function() {
-			var err = new Error('Request cancelled');
-			err.statusCode = 400;
-			onerror(err);
+		xhr.onerror = function(errorEvent) {
+			errorHandler(errorEvent, "XHR error occurred", 8000, 400);
```
Comment (paddybyers) [inline browser/lib/transport/xhrrequest.js]: 80000, not 8000 (and below).

File: browser/lib/transport/xhrrequest.js (hunk 1 above)
Comment (SimonWoolf) [inline browser/lib/transport/xhrrequest.js]: > Also, have we checked that Utils.inspect() gives us something meaningful for the event?

In my tests (just aborting or cutting the connection) it hasn't contained anything useful, no (the stringify was only `{isTrusted: true}`, and the rest of the event object didn't have anything better). Left it in there in the hope that maybe there's some more interesting error might happen in the wild where (a) the event object included some new fields, and (b) they'd be included in the stringify. Perhaps not very likely, but I figure leaving it there doesn't cost much. Want me to remove it?

Comment (paddybyers) [inline browser/lib/transport/xhrrequest.js]: > but I figure leaving it there doesn't cost much. Want me to remove it?

Looking at the event specs you're not going to get anything useful from the event object itself except the type of the event (ie error or abort). TBH I' remove the `inspect()` and just have `'type = ' + ev.type` or something like that in the log message.

**Comment by paddybyers** [comment]
Thanks

**Pull request merged**